internal/common/filetrustsource.go

```diff
@@ -56,27 +61,70 @@ func (source *FileTrustSource) TrustedCertificates() map[string][]*x509.Certific
 	return source.domainCertificates
 }
 
+func (source *FileTrustSource) updateCertificates(certs []*x509.Certificate, spiffeID, path string) error {
+	builder := strings.Builder{}
+	for _, cert := range certs {
+		block := &pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: cert.Raw,
+		}
+		builder.Write(pem.EncodeToMemory(block))
+	}
+
+	file, err := appFS.OpenFile(path, os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	_, err = file.WriteString(builder.String())
+	if err != nil {
+		return err
+	}
+
+	err = source.loadDomain(spiffeID)
+	if err != nil {
+		return err
```

Comment:
Should the certificate file be saved, even if this error occurs?

Reply:
And if so, how will the application recover from this state?

Reply:
Outdated by new design